tests/v2/validation/charts/backup_restore/backup_restore_test.go

```diff
@@ -0,0 +1,204 @@
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/sirupsen/logrus"
+
+	"github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/projects"
+	"github.com/rancher/shepherd/extensions/users"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type BackupTestSuite struct {
+	suite.Suite
+	client              *rancher.Client
+	session             *session.Session
+	project             *management.Project
+	chartInstallOptions *charts.InstallOptions
+	chartFeatureOptions *charts.RancherBackupOpts
+	S3Client            *s3.Client
+}
+
+func (b *BackupTestSuite) TearDownSuite() {
+	b.session.Cleanup()
+}
+
+func (b *BackupTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	b.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(b.T(), err)
+
+	b.client = client
+
+	broConfig := new(charts.Config)
+	config.LoadConfig(charts.ConfigurationFileKey, broConfig)
+
+	project, err := projects.GetProjectByName(client, cluster, clusterProject)
+	require.NoError(b.T(), err)
+
+	// Get clusterName from config yaml
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(b.T(), clusterName, "Cluster name to install is not set")
+
+	// Get cluster meta
+	cluster, err := clusters.NewClusterMeta(client, clusterName)
+	require.NoError(b.T(), err)
+
+	// Get latest version of Rancher Backup chart
+	latestBackupVersion, err := client.Catalog.GetLatestChartVersion(backupChartName, catalog.RancherChartRepo)
+	require.NoError(b.T(), err)
+
+	b.chartInstallOptions = &charts.InstallOptions{
+		Cluster:   cluster,
+		Version:   latestBackupVersion,
+		ProjectID: project.ID,
+	}
+	b.chartFeatureOptions = &charts.RancherBackupOpts{
+		VolumeName:                broConfig.VolumeName,
+		BucketName:                broConfig.S3BucketName,
+		CredentialSecretName:      secretName,
+		CredentialSecretNamespace: broConfig.CredentialSecretNamespace,
+		Enabled:                   backupEnabled,
+		Endpoint:                  broConfig.S3Endpoint,
+		Folder:                    broConfig.S3FolderName,
+		Region:                    broConfig.S3Region,
+	}
+
+	subSession := b.session.NewSession()
+	defer subSession.Cleanup()
+}
+
+func (b *BackupTestSuite) TestBackupChart() {
+	subSession := b.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := b.client.WithSession(subSession)
+	require.NoError(b.T(), err)
+
+	b.client = client
+
+	broConfig := new(charts.Config)
+	config.LoadConfig(charts.ConfigurationFileKey, broConfig)
+
+	project, err := projects.GetProjectByName(b.client, cluster, clusterProject)
+	require.NoError(b.T(), err)
+
+	b.project = project
```

Comment:
I'm not sure I understand this step.. if we are creating a project in the setup why do we have to again get another project and assign it to b.project? Instead can we remove the project from sthe setup and remove it from the testsuite struct and create it here?

Reply:
Good point. I moved almost everything out of the setup and put it into the installBroChart function as that's what most of it was used for.